wasmbinding/test/custom_query_test.go

```diff
@@ -0,0 +1,125 @@
+package wasmbinding_test
+
+import (
+	"encoding/json"
+	"os"
+	"testing"
+
+	"github.com/CosmWasm/wasmd/x/wasm/keeper"
+	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+	"github.com/CosmosContracts/juno/v11/app"
+	appparams "github.com/CosmosContracts/juno/v11/app/params"
+	"github.com/CosmosContracts/juno/v11/x/oracle/wasm"
+	oracle "github.com/CosmosContracts/juno/v11/x/oracle/wasm"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+)
+
+func SetupCustomApp(t *testing.T, addr sdk.AccAddress) (*app.App, sdk.Context) {
+	junoApp, ctx := CreateTestInput(t)
+	wasmKeeper := junoApp.WasmKeeper()
+	storeOracleQuerierCode(t, ctx, junoApp, addr)
+
+	cInfo := wasmKeeper.GetCodeInfo(ctx, 1)
+	require.NotNil(t, cInfo)
+
+	return junoApp, ctx
+}
+
+func TestQueryExchangeRate(t *testing.T) {
+	actor := RandomAccountAddress()
+	junoApp, ctx := SetupCustomApp(t, actor)
+
+	oracleQuerier := instantiateOracleQuerierContract(t, ctx, junoApp, actor)
+	require.NotEmpty(t, oracleQuerier)
+
+	actorAmount := sdk.NewCoins(sdk.NewCoin(appparams.BondDenom, sdk.NewInt(100000000000000)))
+	fundAccount(t, ctx, junoApp, actor, actorAmount)
+
+	ExchangeRateC := sdk.NewDec(1700)
+	ExchangeRateB := sdk.NewDecWithPrec(17, 1)
+	ExchangeRateD := sdk.NewDecWithPrec(19, 1)
+	junoApp.OracleKeeper.SetExchangeRate(ctx, "a", sdk.NewDec(1))
+	junoApp.OracleKeeper.SetExchangeRate(ctx, "b", ExchangeRateB)
+	junoApp.OracleKeeper.SetExchangeRate(ctx, "c", ExchangeRateC)
+	junoApp.OracleKeeper.SetExchangeRate(ctx, "d", ExchangeRateD)
+
+	msg := json.RawMessage(`{"get_exchange_rate": {"denom":"b"}}`)
+	err := executeRawCustom(t, ctx, junoApp, oracleQuerier, actor, msg, sdk.Coin{})
+	require.NoError(t, err)
+
+	query := oracle.OracleQuery{
+		ExchangeRate: &wasm.ExchangeRateQueryParams{
+			Denom: "b",
+		},
+	}
+	resp := oracle.ExchangeRateQueryResponse{}
+	queryRate(t, ctx, junoApp, oracleQuerier, query, &resp)
+
+}
+
+func queryRate(t *testing.T, ctx sdk.Context, junoApp *app.App, contract sdk.AccAddress, request oracle.OracleQuery, response interface{}) {
```

Comment:
This shouldn't be a function on its own, also you should compare the contract state with a query to oraclekeeper instead a hardcode "1.7"